diff -urNp -x '*.orig' grpc-1.32.0.org/CMakeLists.txt grpc-1.32.0/CMakeLists.txt
--- grpc-1.32.0.org/CMakeLists.txt	2020-09-08 23:00:06.000000000 +0200
+++ grpc-1.32.0/CMakeLists.txt	2021-03-04 23:29:28.754072089 +0100
@@ -15666,7 +15666,7 @@ function(generate_pkgconfig name descrip
     "${output_filepath}"
     @ONLY)
   install(FILES "${output_filepath}"
-    DESTINATION "lib/pkgconfig/")
+    DESTINATION "lib${LIB_SUFFIX}/pkgconfig/")
 endfunction()
 
 # gpr .pc file
