diff -urNp -x '*.orig' grpc-1.32.0.org/setup.py grpc-1.32.0/setup.py
--- grpc-1.32.0.org/setup.py	2021-03-04 23:29:27.074067442 +0100
+++ grpc-1.32.0/setup.py	2021-03-04 23:29:27.300734736 +0100
@@ -415,7 +415,7 @@ EXTRAS_REQUIRES = {
 }
 
 SETUP_REQUIRES = INSTALL_REQUIRES + (
-    'Sphinx~=1.8.1',
+    'Sphinx>=1.8.1',
     'six>=1.10',
 ) if ENABLE_DOCUMENTATION_BUILD else ()
 
--- grpc-1.39.0/doc/python/sphinx/conf.py.orig	2021-07-21 00:39:39.000000000 +0200
+++ grpc-1.39.0/doc/python/sphinx/conf.py	2022-02-15 18:47:25.485810968 +0100
@@ -57,7 +57,6 @@ extensions = [
     'sphinx.ext.todo',
     'sphinx.ext.napoleon',
     'sphinx.ext.coverage',
-    'sphinx.ext.autodoc.typehints',
 ]
 
 napoleon_google_docstring = True
